cmd/sos_list.go

```diff
@@ -49,167 +47,114 @@ Note: the buckets size reported is computed daily, it may not be the actual size
+		prefix = prefix + "/"
+	}
 
-			if isShort {
-				fmt.Fprintln(table, key) // nolint: errcheck
-				continue
-			}
+	zone, err := sosClient.GetBucketLocation(bucket)
+	if err != nil {
+		return err
+	}
 
-			fmt.Fprintf(table, "[%s]\t%6s \t%s\n", lastModified, humanize.IBytes(uint64(message.Size)), key) // nolint: errcheck
-		}
+	if err := sosClient.setZone(zone); err != nil {
+		return err
+	}
 
-		return table.Flush()
-	},
+	listObjects(sosClient, bucket, prefix, isRecursive, isShort)
+
+	return nil
 }
 
-func listRecursively(sosClient *sosClient, bucket, prefix, zone string,
-	displayBucket, isShort bool, table io.Writer) {
-	for message := range sosClient.ListObjectsV2(bucket, prefix, true, gContext.Done()) {
-		if message.Err != nil {
-			fmt.Fprintf(os.Stderr, "error: %s\n", message.Err)
-			continue
-		}
+func listObjects(sosClient *sosClient, bucket, prefix string, isRecursive, isShort bool) {
+	table := tabwriter.NewWriter(os.Stdout, 10, 0, 1, ' ', tabwriter.TabIndent)
+
+	for object := range sosClient.ListObjectsV2(bucket, prefix, isRecursive, gContext.Done()) {
+		if object.Err != nil {
+			fmt.Fprintf(os.Stderr, "error: %s\n", object.Err)
 
-		sPrefix := splitPath(prefix)
-		sKey := splitPath(message.Key)
-		if len(prefix) != len(strings.Join(sKey[:len(sPrefix)], "/")) {
+			table.Flush()
 			continue
 		}
 
-		lastModified := message.LastModified.Format(printDate)
-		var bucket string
-		var zoneFormat string
-		if displayBucket {
-			bucket = fmt.Sprintf("%s/", bucket)
-			zoneFormat = fmt.Sprintf("[%s]\t", zone)
-		}
 		if isShort {
-			fmt.Fprintf(table, "%s%s\n", bucket, message.Key) // nolint: errcheck
+			fmt.Fprintf(table, "%s/%s\n", bucket, object.Key) // nolint: errcheck
+
+			table.Flush()
 			continue
 		}
-		fmt.Fprintf(table,
-			"[%s]\t%s%6s \t%s%s\n", lastModified, zoneFormat, humanize.IBytes(uint64(message.Size)), bucket, message.Key) // nolint: errcheck
-	}
-}
-
-func displayBucket(sosClient *sosClient, isRecursive, isShort bool) error {
-	resp, err := cs.RequestWithContext(gContext, egoscale.ListBucketsUsage{})
-	if err != nil {
-		return err
-	}
-
-	buckets := resp.(*egoscale.ListBucketsUsageResponse)
-
-	table := tabwriter.NewWriter(os.Stdout, 0, 0, 1, ' ', tabwriter.TabIndent)
-
-	for _, b := range buckets.BucketsUsage {
-		if isShort {
-			fmt.Fprintf(table, "%s/\n", b.Name) // nolint: errcheck
-		} else {
-			t, err := time.Parse(time.RFC3339, b.Created)
-			if err != nil {
-				return err
-			}
 
+		if object.LastModified.IsZero() {
 			fmt.Fprintf(table,
-				"[%s]\t[%s]\t%6s \t%s/\n", t.Format(printDate), b.Region, humanize.IBytes(uint64(b.Usage)), b.Name) // nolint: errcheck
-		}
-		if isRecursive {
-			if err = sosClient.setZone(b.Region); err != nil {
-				return err
-			}
-			listRecursively(sosClient, b.Name, "", b.Region, true, isShort, table)
+				"%s\t%s\t%s/%s\t\n",
+				strings.Repeat(" ", 25),
+				"DIR",
+				bucket, object.Key) // nolint: errcheck
+
+			table.Flush()
```

Comment:
I think you can set one call to `table.Flush()` at the beginning of the loop ;) instead of setting it before each `continue` keyword.